spec/Functions.tex

```diff
@@ -1041,6 +1041,37 @@ \subsection{Implicit Return Types}
 inferred return type.
 If the above requirements are not satisfied, it is an error.
 
+\section{Where Expressions}
+\label{Where_Expressions}
+\index{where@\chpl{where}}
+\index{expressions!where@\chpl{where}}
+\index{functions!where@\chpl{where}}
+
+The list of function candidates can be constrained by {\em where clauses}.  A
+where clause is specified in the definition of a
+function~(\rsec{Function_Definitions}).  When a function is resolved, the
+expression in the where clause must be a parameter expression and must
+evaluate to either \chpl{true} or \chpl{false}.  If it evaluates to
+\chpl{false}, the function is rejected and thus is not a possible candidate
+for function resolution.
+\begin{chapelexample}{whereClause.chpl}
+Given two overloaded function definitions
+\begin{chapel}
+proc foo(x) where x.type == int { writeln("int"); }
+proc foo(x) where x.type == real { writeln("real"); }
+\end{chapel}
```

Comment:
OK, just wanted to make sure this wasn't the only such case.  If we already do this, I like this approach better than the alternative I proposed and think this could go in either as-is, or taking the wordsmithing suggestion above.